shell/browser/api/atom_api_screen.cc

```diff
@@ -99,17 +99,38 @@ static gfx::Rect DIPToScreenRect(electron::NativeWindow* window,
 
 #endif
 
+void DelayEmit(Screen* screen,
+               const base::StringPiece& name,
+               const display::Display& display) {
+  screen->Emit(name, display);
+}
+
+void DelayEmitWithMetrics(Screen* screen,
+                          const base::StringPiece& name,
+                          const display::Display& display,
+                          std::vector<std::string> metrics) {
+  screen->Emit(name, display, metrics);
+}
+
 void Screen::OnDisplayAdded(const display::Display& new_display) {
-  Emit("display-added", new_display);
+  base::PostTaskWithTraits(FROM_HERE, {content::BrowserThread::UI},
```

Comment:
you can post to the current thread like
```cpp
base::ThreadTaskRunnerHandle::Get()->PostTask(...)
```

but i guess we're guaranteed to be on the ui thread here anyway so /shrug